File: clients/apple/MultiLiveTV/Services/NetworkConfig.swift
import Foundation

enum NetworkConfig {
    static let userAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
    static let requestTimeout: TimeInterval = 15
}

enum RequestFailure {
    static func userFacingMessage(for error: Error) -> String {
        if let urlError = error as? URLError {
            return message(for: urlError.code)
        }
        let nsError = error as NSError
        if nsError.domain == NSURLErrorDomain {
            return message(for: URLError.Code(rawValue: nsError.code))
        }
        if let localized = error as? LocalizedError,
           let description = localized.errorDescription,
           !description.isEmpty {
            return description
        }
        return "加载失败，请稍后重试"
    }

    private static func message(for code: URLError.Code) -> String {
        switch code {
        case .timedOut:
            return "请求超时，请检查网络后重试"
        case .notConnectedToInternet, .dataNotAllowed:
            return "网络不可用，请检查网络后重试"
        case .cannotFindHost, .dnsLookupFailed:
            return "无法连接服务器，请稍后重试"
        case .cannotConnectToHost, .networkConnectionLost, .cannotLoadFromNetwork:
            return "连接失败，请稍后重试"
        case .secureConnectionFailed, .serverCertificateUntrusted, .serverCertificateHasBadDate,
             .serverCertificateHasUnknownRoot, .serverCertificateNotYetValid, .clientCertificateRejected,
             .clientCertificateRequired:
            return "安全连接失败，请稍后重试"
        case .badServerResponse, .zeroByteResource, .cannotParseResponse, .cannotDecodeContentData,
             .cannotDecodeRawData:
            return "服务器响应异常，请稍后重试"
        case .badURL, .unsupportedURL:
            return "请求地址无效"
        default:
            return "加载失败，请稍后重试"
        }
    }
}

enum RemoteMediaURL {
    static func parse(_ string: String) -> URL? {
        var trimmed = string.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return nil }
        if trimmed.hasPrefix("//") {
            trimmed = "https:" + trimmed
        }
        guard let url = makeURL(trimmed), isUsable(url) else { return nil }
        return url
    }

    private static func makeURL(_ string: String) -> URL? {
        if let url = URL(string: string) { return url }
        return string.addingPercentEncoding(withAllowedCharacters: .urlFragmentAllowed).flatMap(URL.init(string:))
    }

    private static func isUsable(_ url: URL) -> Bool {
        guard let scheme = url.scheme?.lowercased(), scheme == "http" || scheme == "https" else {
            return false
        }
        return url.host?.isEmpty == false
    }
}

File: clients/android/core/src/main/kotlin/com/heibaimiao/multilivetv/net/NetworkConfig.kt
package com.heibaimiao.multilivetv.net

object NetworkConfig {
    const val USER_AGENT =
        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
    const val CONNECT_TIMEOUT_SECONDS = 5L
    const val REQUEST_TIMEOUT_SECONDS = 8L
}

object RemoteMediaURL {
    fun parse(string: String): String? {
        var trimmed = string.trim()
        if (trimmed.isEmpty()) return null
        if (trimmed.startsWith("//")) trimmed = "https:$trimmed"
        return if (isUsable(trimmed)) trimmed else null
    }

    private fun isUsable(url: String): Boolean {
        val lower = url.lowercase()
        if (!lower.startsWith("http://") && !lower.startsWith("https://")) return false
        val withoutScheme = url.substringAfter("://")
        val host = withoutScheme.substringBefore("/").substringBefore("?")
        return host.isNotEmpty()
    }
}

object RequestFailure {
    fun userFacingMessage(error: Throwable): String {
        val message = error.message.orEmpty().lowercase()
        return when {
            "timeout" in message || "timed out" in message -> "请求超时，请检查网络后重试"
            "unable to resolve" in message || "unknownhost" in message -> "无法连接服务器，请稍后重试"
            "connect" in message || "connection" in message -> "连接失败，请稍后重试"
            "ssl" in message || "certificate" in message || "tls" in message -> "安全连接失败，请稍后重试"
            "http" in message || "unexpected code" in message -> "服务器响应异常，请稍后重试"
            error.message?.isNotBlank() == true && error is VodClientException -> error.message!!
            else -> "加载失败，请稍后重试"
        }
    }
}

class VodClientException(message: String) : Exception(message)
